### DISCUSSION TITLE
filepath.c:2162:27: warning: writing 1 byte into a region of size 0

### DISCUSSION THREAD

**Discussion by independentCog** [vim/vim#13352] [Q&A]
I was able to successfully compile from Vim src but received this sole warning:

`gcc -c -I. -Iproto -DHAVE_CONFIG_H     -g -O2 -D_REENTRANT -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1        -o objects/filepath.o filepath.c
filepath.c: In function ‘f_resolve’:
filepath.c:2162:27: warning: writing 1 byte into a region of size 0 [-Wstringop-overflow=]
 2162 |                     q[-1] = NUL;` 
 
 Used cygwin, and here's my config:
 `./configure --with-features=huge --enable-python3interp --with-python3-config-dir=/usr/bin/python3.9 --with-python3-config-dir=/usr/include/python3.9 --enable-cscope --enable-fail-if-missing --enable-luainterp=yes --enable-terminal --prefix="/cygdrive/e/compile/"` In case it's somehow related.
 
 The code in question at src/filepath.c seems to ignore a trailing path separator by setting the character before it to null. My understanding of C isn't great so I could be wrong about it. This warning could be trivial because I was able to make install it successfully. Ultimately was just curious what the implications of this are and what this error is related to?

**Comment by k-takata** [comment]
I think this is a false positive, because there is a check `q > p` in line 2159.
However, we can suppress the warning by the following patch:

```diff
--- a/src/filepath.c
+++ b/src/filepath.c
@@ -2159,7 +2159,7 @@ f_resolve(typval_T *argvars, typval_T *r
 		if (q > p && *q == NUL)
 		{
 		    // Ignore trailing path separator.
-		    q[-1] = NUL;
+		    p[q - p - 1] = NUL;
 		    q = gettail(p);
 		}
 		if (q > p && !mch_isFullName(buf))
```

**Reply by k-takata** [reply]
I've confirmed that Cygwin GCC generates exactly the same code with and without the patch.

**Comment by k-takata** [comment]
Fixed by https://github.com/vim/vim/commit/215c3261a25f7a99e8711a3b3c6158119c6aea9e. Closing.
